contributor-docs/adrs/adr-020-live-regions.md

```diff
@@ -0,0 +1,165 @@
+# ADR 020: Live Regions
+
+## Status
+
+| Stage    | Status      |
+| -------- | ----------- |
+| Approved | <!-- ✅ --> |
+| Adopted  | <!-- 🚧 --> |
+
+## Context
+
+There are several components in Primer React that make use of live regions.
+However, the library does not have a central pattern for making accessible live
+region announcements. This ADR documents the decision to use a central pattern
+for live region announcements that can be used across Primer and GitHub.
+
+ARIA Live regions fall into several buckets that we may want to use as component
+authors:
+
+- [ARIA live region roles](https://www.w3.org/TR/wai-aria-1.2/#live_region_roles), such as `alert`, `log`, or `status`
+- [The `aria-live` attribute](https://www.w3.org/TR/wai-aria-1.2/#aria-live) to turn an element into a live region
+
+In components, we see the following scenarios in Primer React:
+
+- Announce the contents of an element when it is rendered or on page load, like when a spinner is displayed or a form is submitted
+- Announce the changes to the content of an element, like when the count inside of a
+  button is incremented
+- Announce a message programmatically, such as the number of results for a query
+
+Currently, contributors may reach for roles such as `alert` or `status` to
+achieve these scenarios. They may also add `aria-live` to an element explicitly.
+However, both of these approaches do not announce consistently across screen
+readers.
+
+### Links & Resources for ARIA Live regions
+
+- https://www.sarasoueidan.com/blog/accessible-notifications-with-aria-live-regions-part-1/
+- https://www.scottohara.me/blog/2022/02/05/are-we-live.html
+
+## Decision
+
+In order to have a common interop point for live region announcents, Primer React will
+make use of a `live-region` custom element from `@primer/live-region-element`.
+This package will be included and published from the `primer/react` repo.
+
+The custom element exposes a way to make announcements that can be used
+across frameworks. This makes it useful not only for Primer but GitHub as a
+whole.
+
+In addition, `@primer/react` will leverage and export the following helpers for
+use within Primer React and GitHub:
+
+- `announce()` and `announceFromElement()`, programmatic helpers for making
```

Comment:
@khiga8 I think groups outside of Primer React could use `@primer/live-region-element` directly! This would just be as a convenience, but happy to make it so that folks only import/use `@primer/live-region-element` if that would be clearer.

Reply:
Yes I think that would make it more clear :)